core/src/main/java/com/turn/ttorrent/common/Torrent.java

```diff
@@ -676,21 +676,25 @@ private static Torrent create(File parent, List<File> files, int pieceLength,
 
 		private final MessageDigest md;
 		private final ByteBuffer data;
+		private final ByteBufferPool bbp;
 
-		CallableChunkHasher(ByteBuffer buffer) throws NoSuchAlgorithmException {
+		CallableChunkHasher(ByteBuffer rentedBuffer, ByteBufferPool bbp) throws NoSuchAlgorithmException {
```

Comment:
You have an asymmetry in the responsibility of dealing with the buffer pool that's concerning here: this class doesn't seem responsible for acquiring the buffer from the pool, but is responsible for returning it. That's usually a code smell.